src/Akeneo/Category/back/Application/Storage/Save/CategorySaverRegistry.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Category\Application\Storage\Save;
+
+use Akeneo\Category\Application\Storage\Save\Saver\CategorySaver;
+use Webmozart\Assert\Assert;
+
+/**
+ * This class returns a category data saver based on a user intent.
+ *
+ * @copyright 2022 Akeneo SAS (https://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class CategorySaverRegistry
+{
+    /** @var array<string, CategorySaver>  */
+    private array $categorySaverByUserIntent;
+
+    /**
+     * @param CategorySaver[] $categorySavers
+     */
+    public function __construct(
+        private array $categorySavers
+    ) {
+        foreach ($this->categorySavers as $categorySaver) {
```

Comment:
There is a lot of logic in the constructor, maybe can we extract this part in a private function ? 
```suggestion
        $this->assertUniqueIntentSavers()
```

Reply:
good point

Reply:
done